File: mobile/app/src/main/java/com/example/phoneoff/Model/Product.java
package com.example.phoneoff.Model;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Base64;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "Product")
public class Product {
    @PrimaryKey
    public int Id;
    public String Name;
    public double Cost;
    public String Description;
    public String Color;
    public String Manufacturer;
    public int RAM;
    public int ROM;
    public double Diagonal;
    public String Image;
    public int Count;

    public Product() {

    }

    public Product(int id, String name, double cost, String description, String color, String manufacturer, int RAM, int ROM, double diagonal, String image, int count) {
        Id = id;
        Name = name;
        Cost = cost;
        Description = description;
        Color = color;
        Manufacturer = manufacturer;
        this.RAM = RAM;
        this.ROM = ROM;
        Diagonal = diagonal;
        Image = image;
        Count = count;
    }

    public int getId() {
        return Id;
    }

    public void setId(int id) {
        Id = id;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public double getCost() {
        return Cost;
    }

    public void setCost(double cost) {
        Cost = cost;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public String getColor() {
        return Color;
    }

    public void setColor(String color) {
        Color = color;
    }

    public String getManufacturer() {
        return Manufacturer;
    }

    public void setManufacturer(String manufacturer) {
        Manufacturer = manufacturer;
    }

    public int getRAM() {
        return RAM;
    }

    public void setRAM(int RAM) {
        this.RAM = RAM;
    }

    public int getROM() {
        return ROM;
    }

    public void setROM(int ROM) {
        this.ROM = ROM;
    }

    public double getDiagonal() {
        return Diagonal;
    }

    public void setDiagonal(double diagonal) {
        Diagonal = diagonal;
    }

    public void setImage(String image) {
        Image = image;
    }

    public Bitmap getImage() {
        return StringToBitMap(Image);
    }

    public int getCount() {
        return Count;
    }

    public void setCount(int count) {
        Count = count;
    }

    private Bitmap StringToBitMap(String encodedString) {
        try {
            byte[] encodeByte = Base64.decode(encodedString, Base64.DEFAULT);
            Bitmap bitmap = BitmapFactory.decodeByteArray(encodeByte, 0, encodeByte.length);
            return bitmap;
        } catch (Exception e) {
            e.getMessage();
            return null;
        }
    }

}

File: WebApplication/Product.cs
using System;
using System.Data;

namespace WebApplication
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Cost { get; set; }
        public string Description { get; set; }
        public string Color { get; set; }
        public string Manufacturer { get; set; }
        public int ROM { get; set; }
        public int RAM { get; set; }
        public double Diagonal { get; set; }
        public byte[] Image { get; set; }
        public int Count { get; set; }

        public Product(int id, string name, double cost, string description, string color, string manufacturer, int rOM, int rAM, double diagonal, byte[] image, int count)
        {
            Id = id;
            Name = name;
            Cost = cost;
            Description = description;
            Color = color;
            Manufacturer = manufacturer;
            ROM = rOM;
            RAM = rAM;
            Diagonal = diagonal;
            Image = image;
            Count = count;
        }

        public Product(DataRow row)
        {
            Id = int.Parse(row[0].ToString());
            Name = row[1].ToString();
            Description = row[2].ToString();
            Cost = int.Parse(row[3].ToString());
            Image = (byte[]) row[4];
            Manufacturer = (string) row[5];
            Color = (string) row[6];
            ROM = int.Parse(row[7].ToString());
            RAM = int.Parse(row[8].ToString());
            Diagonal = double.Parse(row[9].ToString());
            Count = int.Parse(row[10].ToString());
        }
    }


    public class ProductWithoutId
    {
        public string Name { get; set; }
        public double Cost { get; set; }
        public string Description { get; set; }
        public int Color { get; set; }
        public int Manufacturer { get; set; }
        public int ROM { get; set; }
        public int RAM { get; set; }
        public int Diagonal { get; set; }
        public byte[] Image { get; set; }
        public int Count { get; set; }

        public ProductWithoutId(string name, double cost, string description, int color, int manufacturer, int rOM, int rAM, int diagonal, byte[] image, int count)
        {
            Name = name;
            Cost = cost;
            Description = description;
            Color = color;
            Manufacturer = manufacturer;
            ROM = rOM;
            RAM = rAM;
            Diagonal = diagonal;
            Image = image;
            Count = count;
        }
    }


    public class ProductUpdate
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public double Cost { get; set; }
        public string Description { get; set; }
        public int Color { get; set; }
        public int Manufacturer { get; set; }
        public int ROM { get; set; }
        public int RAM { get; set; }
        public int Diagonal { get; set; }
        public byte[] Image { get; set; }
        public int Count { get; set; }

        public ProductUpdate(int id, string name, double cost, string description, int color, int manufacturer, int rOM, int rAM, int diagonal, byte[] image, int count)
        {
            Id = id;
            Name = name;
            Cost = cost;
            Description = description;
            Color = color;
            Manufacturer = manufacturer;
            ROM = rOM;
            RAM = rAM;
            Diagonal = diagonal;
            Image = image;
            Count = count;
        }
    }

    public class ProductCount
    {
        public int Id { get; set; }
        public int Count { get; set; }

        public ProductCount(int id, int count)
        {
            Id = id;
            Count = count;
        }
    }
}